YoutubeExplode/Internal/YoutubeVideoStream.cs

```diff
@@ -0,0 +1,115 @@
+using System;
+using System.IO;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace YoutubeExplode.Internal
+{
+    internal class YoutubeVideoStream : Stream
+    {
+        private readonly HttpClient _httpClient;
+
+        private const int MaxSegmentSize = 9_898_989;
+
+        private readonly string _url;
+
+        Stream _currentStream;
+
+        private long _position;
+
+        public YoutubeVideoStream(HttpClient httpClient, string url, long length)
+        {
+            _url = url;
+            _httpClient = httpClient;
+            Length = length;
+        }
+
+        public override bool CanRead => true;
+
+        public override bool CanSeek => true;
+
+        public override bool CanWrite => false;
+
+        public override long Length { get; }
+
+        public override long Position
+        {
+            get => _position;
+            set
+            {
+                if (value < 0)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(value), "Non-negative number required");
+                }
+                _position = value;
+            }
+        }
+
+        public override void Flush() => throw new System.NotSupportedException();
+
+        public override async Task<int> ReadAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
+        {
+            if (Position >= Length)
+                return 0;
+
+            if (_currentStream == null)
+                _currentStream = await _httpClient.GetStreamAsync(_url, Position, Position + MaxSegmentSize).ConfigureAwait(false);
+
+            var bytesRead = await _currentStream.ReadAsync(buffer, offset, count, cancellationToken).ConfigureAwait(false);
+            Position += bytesRead;
+            if (bytesRead == 0)
```

Comment:
When is this condition met?

Reply:
When current Stream from Youtube ends, just like with [AsyncMultiStream](https://github.com/Tyrrrz/YoutubeExplode/pull/162/files#diff-0d3b577d11ab6846312caf4545e1ebcbL41)